Source/core/MessageException.h

```diff
@@ -21,23 +21,43 @@
 #define EXCEPTION_H
 
 #include <exception> // For exception class
+#include <errno.h>
 
 #include "Module.h"
 #include "Portability.h"
 #include "TextFragment.h"
+#include "Serialization.h"
 
 namespace Thunder {
 namespace Core {
 
-    class MessageException : public std::exception {
-    private:
-        MessageException();
-
+    class  MessageException : public std::exception {
     public:
-        MessageException(const string& message, bool inclSysMsg = false) throw();
-        ~MessageException() throw();
+        MessageException() = delete;
+        MessageException(const MessageException&) = delete;
+        MessageException(MessageException&&) = delete;
+        MessageException& operator=(const MessageException&) = delete;
+        MessageException& operator=(MessageException&&) = delete;
+
+PUSH_WARNING(DISABLE_WARNING_DEPRECATED_USE)
+        inline MessageException(const string& message, bool inclSysMsg) noexcept(true)
```

Comment:
This is a bit tricky, You are changing the interface (breaking it).The original interface could work with only 1 parameter (hence the explicit request on the constructor of this). Now it is mandatory that the constructor must have a boolean as the second parameter, which means that if there is code that is using this (might be in RDKServices) you break it with this change. Suggest to keep the signature of the constructor the same and make the constructor  explicit. Unless there is a good reason why you want to break it (technical issues ?)
P.S. You can argue if the explicit could cause a break as well, since you now need to do an explicit converion. Might also break the code but at least here the technical argument would be that it makes the code more pure, avoids implicit, unwanted conversions and object creations, the developper was not aware of..

Reply:
Just missed the default value in the C&P.